pkg/controllers/machine_controller.go

```diff
@@ -173,6 +189,118 @@ func (r *MachineReconciler) Start(ctx context.Context) error {
 	return nil
 }
 
+func (r *MachineReconciler) startGarbageCollector(ctx context.Context, log logr.Logger) {
+	wait.UntilWithContext(ctx, func(ctx context.Context) {
+		machines, err := r.machines.List(ctx)
```

Comment:
i think we need recovery logic for panic. this function has to be design as failure tolerant

Reply:
recovery logic is part of wait.UntilWithContext. it little looks behavior of panic propagation to top is manage be variable ReallyCrash which is set to true by default: https://github.com/kubernetes/apimachinery/blob/master/pkg/util/runtime/runtime.go#L34

Reply:
minimally panic has to be logged in same format as other log records.

garbage collector is for me core functionality of libvirt-provider which has to work. But i prefer here really failure tolerant logic. Main reason is affect another more important logic (create machine). When machine cannot be deleted okay it is problem but you can still create new machines and you have time solve root cause with GC. If panic crash application, app can be stuck in infinity restart loop and i think it is worse case.

Reply:
As discussed privately, there is no need for us to be concerned about this at the moment.